### PR TITLE
Fixes #32083 - use full Host model in compute

### PR DESCRIPTION
**Pull request by ezr-ondrej**
For rendering `hosts/_compute` we've used just host attributes in the
past, but it just brings more complexity, when we cant rely on the Host
model to be around.

This show cases how we would work with documentation presented in #8391

### CODE REVIEW

**Comment by lzap** [comment]
I am confused, the RM issue states this is some documentation effort however I do see code changes and the short description does not deliver for me. Can you just a bit elaborate this?

**Comment by ezr-ondrej** [comment]
[test katello]

**Review by lzap — APPROVED** [review]
Works fine with libvirt.

File: app/helpers/hosts_helper.rb [hunk 1]
```diff
@@ -304,14 +304,6 @@ def has_dhcp_lease_errors?(errors)
     errors.include?(:dhcp_lease_error)
   end
 
-  def args_for_compute_resource_partial(host)
-    args = {}
-    args[:arch] = host.try(:architecture_id) || (params[:host] && params[:host][:architecture_id])
-    args[:os] = host.try(:operatingsystem_id) || (params[:host] && params[:host][:operatingsystem_id])
-    args[:selected_cluster] = vm_attrs['cluster'] if defined?(vm_attrs)
-    args
-  end
```
Comment (tbrisker) [inline app/helpers/hosts_helper.rb:313]: FYI: This change broke the proxmox plugin (see https://github.com/theforeman/foreman_fog_proxmox/issues/199) and potentially others. In the future, please make sure to deprecate methods before removing them completely.

**Pull request merged**